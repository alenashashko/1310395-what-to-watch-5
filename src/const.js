export const MillisecondsInTimePeriod = {
  SECOND: 1000,
  MINUTE: 60000,
  HOUR: 3600000,
};
