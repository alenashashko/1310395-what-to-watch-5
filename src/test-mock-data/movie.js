export const movie = {
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
};

export const movies = [{
  id: `1`,
  title: `Orlando`,
  picture: `/img/orlando.jpg`,
  backgroundPicture: `/img/orlando.jpg`,
  backgroundColor: `#e1b0b2`,
  poster: `/img/orlando.jpg`,
  genre: `Crime`,
  year: 2015,
  ratingScore: 7.8,
  ratingCount: 1000,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 110,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.
  Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.
  Sed sed nisi sed augue convallis suscipit in sed felis.
  Aliquam erat volutpat.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`,
    `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`],
  isFavorite: false
},
{
  id: `2`,
  title: `Revenant`,
  picture: `/img/revenant.jpg`,
  backgroundPicture: `/img/revenant.jpg`,
  backgroundColor: `#bdad8f`,
  poster: `/img/revenant.jpg`,
  genre: `Romance`,
  year: 2010,
  ratingScore: 5.9,
  ratingCount: 850,
  src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  duration: 145,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  Cras aliquet varius magna, non porta ligula feugiat eget.
  Fusce tristique felis at fermentum pharetra.
  Aliquam id orci ut lectus varius viverra.
  Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.
  Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
  director: `Robert Zemeckis`,
  starring: [`Will Smith`, `Michael J. Fox`, `Harrison Ford`,
    `Sandra Bullock`, `Betty White`, `Tom Hanks`, `Denzel Washington`],
  isFavorite: false
}
];

export const genres = [`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`,
  `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`];

export const genre = `All genres`;
