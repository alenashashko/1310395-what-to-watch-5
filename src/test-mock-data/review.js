import {adaptCommentToClient} from '../services/adapters';

export const apiReviews = [
  {
    id: 1,
    user: {
      id: 11,
      name: `Jack`
    },
    rating: 2.4,
    comment: `It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.`,
    date: `2020-11-03T13:38:44.769Z`
  },
  {
    id: 2,
    user: {
      id: 12,
      name: `Isaac`
    },
    rating: 7.2,
    comment: `A movie that will take you to another world full of emotions.`,
    date: `2020-10-23T13:38:44.769Z`
  }
];

export const review = adaptCommentToClient(apiReviews[0]);

export const reviews = apiReviews.map(adaptCommentToClient);

