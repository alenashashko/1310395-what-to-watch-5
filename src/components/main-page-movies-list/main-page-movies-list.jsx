import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import proptypes from '../../type';
import MoviesList from '../movies-list/movies-list';
import ShowMoreButton from '../show-more-button/show-more-button';
import {MOVIES_PER_PAGE} from '../../const';
import withPageNumber from '../../hocs/with-page-number/with-page-number';

const MainPageMoviesList = (props) => {
  const {movies, pageNumber, onButtonClick} = props;

  const getVisibleMovies = () => {
    return movies.slice(0, pageNumber * MOVIES_PER_PAGE);
  };

  const visibleMovies = getVisibleMovies();

  return (
    <React.Fragment>
      <MoviesList movies={visibleMovies} />

      {visibleMovies.length < movies.length ?
        <ShowMoreButton onButtonClick={onButtonClick} /> :
        null}
    </React.Fragment>
  );
};

MainPageMoviesList.propTypes = {
  movies: proptypes.movies,
  pageNumber: proptypes.pageNumber,
  onButtonClick: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  movies: state.moviesByGenre
});

export default connect(mapStateToProps)(withPageNumber(MainPageMoviesList));
