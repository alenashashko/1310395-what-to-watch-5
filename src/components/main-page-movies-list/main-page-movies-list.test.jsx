import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';

import {MainPageMoviesList} from './main-page-movies-list';
import {DEFAULT_MOVIES_FILTER_VALUE} from '../../const';
import {movies} from '../../test-mock-data/movie';

const visibleMovies = movies.slice(0, 1);

const noop = () => {};

describe(`Should MainPageMoviesList render correctly`, () => {
  it(`when movies.length and visibleMovies.length are equal`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <MainPageMoviesList
              movies={movies}
              visibleMovies={movies}
              onButtonClick={noop}
              resetPageNumber={noop}
              currentGenre={DEFAULT_MOVIES_FILTER_VALUE}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`when movies.length more than visibleMovies.length`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <MainPageMoviesList
              movies={movies}
              visibleMovies={visibleMovies}
              onButtonClick={noop}
              resetPageNumber={noop}
              currentGenre={DEFAULT_MOVIES_FILTER_VALUE}
            />
          </BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
