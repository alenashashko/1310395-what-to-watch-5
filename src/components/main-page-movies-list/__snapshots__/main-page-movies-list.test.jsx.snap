// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainPageMoviesList render correctly when movies.length and visibleMovies.length are equal 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Bronson"
        height="175"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/1"
        onClick={[Function]}
      >
        Bronson
      </a>
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Pulp Fiction"
        height="175"
        src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="/films/2"
        onClick={[Function]}
      >
        Pulp Fiction
      </a>
    </h3>
  </article>
</div>
`;

exports[`Should MainPageMoviesList render correctly when movies.length more than visibleMovies.length 1`] = `
Array [
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <img
          alt="Bronson"
          height="175"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/1"
          onClick={[Function]}
        >
          Bronson
        </a>
      </h3>
    </article>
  </div>,
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>,
]
`;
